TLM/TLM/U/UIScaler.cs

```diff
@@ -5,11 +5,46 @@ namespace TrafficManager.U {
     using UnityEngine;
 
     public static class UIScaler {
+        public const float GUI_WIDTH = 1920f;
+        public const float GUI_HEIGHT = 1080f;
+
+        /// <summary>Caching because UIView.Instance.uiCamera can be null sometimes.</summary>
+        private static float cachedGuiWidth = GUI_WIDTH;
+
+        /// <summary>Caching because UIView.Instance.uiCamera can be null sometimes.</summary>
+        private static float cachedGuiHeight = GUI_HEIGHT;
+
         /// <summary>Screen width for GUI is always fixed at 1920.</summary>
-        public static float GuiWidth => Singleton<UIView>.instance.uiCamera.pixelWidth;
+        public static float GuiWidth {
+            // TODO: Double check if GUI never changes width, the code below can be a const
+            get {
+                UIView uiView = Singleton<UIView>.instance;
+                if (uiView != null) {
+                    Camera uiCamera = uiView.uiCamera;
```

Comment:
just a minor note.  it would be more compact to write like this:
```c#
UIView uiView = Singleton<UIView>.instance;
Camera uiCamera = uiView?.uiCamera;
if(uiCamera !=null) 
  ...
```

Reply:
But this does not eliminate the null check, in your code null check will happen twice.

Reply:
yeah its the same code just more compact.

Reply:
Code compactness is nice, but not at the expense of obscuring clarity of intent of the code.